pkg/controller/core/workload_controller.go

```diff
@@ -830,11 +830,7 @@ func (r *WorkloadReconciler) admittedNotReadyWorkload(wl *kueue.Workload) (bool,
 	if podsReadyCond != nil && podsReadyCond.Status == metav1.ConditionFalse && podsReadyCond.LastTransitionTime.After(admittedCond.LastTransitionTime.Time) {
 		elapsedTime = r.clock.Since(podsReadyCond.LastTransitionTime.Time)
 	}
-	waitFor := r.waitForPodsReady.timeout - elapsedTime
-	if waitFor < 0 {
-		waitFor = 0
-	}
-	return true, waitFor
+	return true, r.waitForPodsReady.timeout - elapsedTime
```

Comment:
As I checked the affection by this change, this change does not have any bug for now since we use this time only if the time is positive number:
https://github.com/kubernetes-sigs/kueue/blob/7a496666ec137cc08315df1884de623e0d68cd2a/pkg/controller/core/workload_controller.go#L546-L566

However, in the future, we do not have any guarantee. So, I would recommend using the `max` function as @mimowo mentioned above.

```suggestion
	return true, max(r.waitForPodsReady.timeout - elapsedTime, 0)
```

Reply:
Done